pkg/signing/signing.go

```diff
@@ -274,6 +275,51 @@ func getSigsPrimPart(fimg *sif.FileImage) (sigs []*sif.Descriptor, descr []*sif.
 	return
 }
 
+func getSigsAllPart(fimg *sif.FileImage) ([]*sif.Descriptor, []*sif.Descriptor, error) {
+	var dpart []*sif.Descriptor
+	var sigs []*sif.Descriptor
+	var err error
+
+	// Ensure theres a primary partition.
+	_, _, err = fimg.GetPartPrimSys()
+	if err != nil {
+		return nil, nil, fmt.Errorf("no primary partition found")
+	}
+
+	// Loop thrugh all the partitions, (skipping DataSignatures)
+	// and collect all the signatures for a data partition.
+	for _, d := range fimg.DescrArr {
+		if !d.Used {
+			continue
+		}
+		// No need to verify a signature.
+		if d.Datatype == sif.DataSignature {
+			continue
+		}
+		s, _, err := fimg.GetLinkedDescrsByType(d.ID, sif.DataSignature)
+		if err != nil {
+			// If a partition is not signed, print a warning.
+			// TODO: how will/should this efect the exit code?
```

Comment:
The exit codes for verify are something I think we need to consider at a higher level in general and document. This is a bit off topic, but what are the current exit codes in use?

Reply:
Currently, by default (`singularity verify foo.sif`), the exit code will be 0 if there is no corrupted data. If the `--local` flag is specified, them it will succeed only if all the keys are local.

Actually, I think the current behavior for this PR is fine. If a user wants to ensure a specific partition is signed, then they need to use the `--sif-id` flag.